src/graphics.rs

```diff
@@ -1704,19 +1752,24 @@ impl Buffer {
             gl_buf,
             buffer_type,
             size,
+            index_type: None,
         }
     }
 
-    pub fn update<T>(&self, ctx: &mut Context, data: &[T]) {
+    pub fn update<T>(&mut self, ctx: &mut Context, data: &[T]) {
```

Comment:
Unfortunately `update` can't have `&mut self`. 

Buffer is a copy-able handle for OpenGl buffer. Modifying one of the copies will not necessary modify the one that will be lately used for drawing :( 

The idea here - we can set up some fields in a Buffer struct during the initialization, but later on, the Buffer struct is just an ID